akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala

```diff
@@ -148,9 +148,13 @@ private[stream] object InputStreamSinkStage {
   override def read(a: Array[Byte], begin: Int, length: Int): Int = {
     require(a.length > 0, "array size must be >= 0")
     require(begin >= 0, "begin must be >= 0")
-    require(length > 0, "length must be > 0")
+    require(length >= 0, "length must be >= 0")
     require(begin + length <= a.length, "begin + length must be smaller or equal to the array length")
 
+    if (length == 0) {
+      return 0
```

Comment:
Return in Scala is implemented with some unpleasant trixery under the hood, better do this as an 

```
if (length == 0) 0
else {
...
}
```

Reply:
Changed it to `if ... else ...` without braces